lib/src/execute.rs

```diff
@@ -84,7 +84,7 @@ impl ExecuteCtx {
         module_path: impl AsRef<Path>,
         profiling_strategy: ProfilingStrategy,
         wasi_modules: HashSet<ExperimentalModule>,
-        guest_profile_path: Option<PathBuf>,
+        guest_profile_path: Option<String>,
```

Comment:
I think we can avoid this breaking change in the viceroy_lib crate by applying this change instead:
```diff
diff --git a/cli/src/main.rs b/cli/src/main.rs
index bcf7524..db46cdf 100644
--- a/cli/src/main.rs
+++ b/cli/src/main.rs
@@ -240,7 +240,7 @@ async fn create_execution_context(
         input,
         args.profiling_strategy(),
         args.wasi_modules(),
-        guest_profile_path,
+        guest_profile_path.map(|p| p.into()),
         args.unknown_import_behavior(),
     )?
     .with_log_stderr(args.log_stderr())
diff --git a/lib/src/execute.rs b/lib/src/execute.rs
index bb45d86..df731fe 100644
--- a/lib/src/execute.rs
+++ b/lib/src/execute.rs
@@ -84,7 +84,7 @@ impl ExecuteCtx {
         module_path: impl AsRef<Path>,
         profiling_strategy: ProfilingStrategy,
         wasi_modules: HashSet<ExperimentalModule>,
-        guest_profile_path: Option<String>,
+        guest_profile_path: Option<PathBuf>,
         unknown_import_behavior: UnknownImportBehavior,
     ) -> Result<Self, Error> {
         let config = &configure_wasmtime(profiling_strategy);
@@ -112,7 +112,6 @@ impl ExecuteCtx {
                 engine_clone.increment_epoch();
             }
         })));
-        let guest_profile_path = guest_profile_path.map(PathBuf::from);
 
         Ok(Self {
             engine,
```

Reply:
I've updated this PR to deal with the breaking change.